elastic/elastic-agent
# Add auto diagnostic collection on test failure and ability to skip uninstall for inspection of installed agent

## What does this PR do?



This provides the automatic collection of diagnostics when a tests fails in the integration testing suite. Including providing them as artifacts to buildkite so they can be used.

This also includes 2 environment variable options to help debug using the integration testing framework.

First one is the `AGENT_COLLECT_DIAG=true` option that instructs the testing fixture to collect the diagnostics of the Elastic Agent that is installed before performing the uninstall. (Doesn't have to be a failure for the collection)

Second one is the `AGENT_KEEP_INSTALLED=true` option that prevents the uninstallation of the Elastic Agent at the end of the test. Should only be used with the `mage integration:single [testName]` but helps with the ability to debug the still running Elastic Agent. With the latest change of not bring down the instances at the end, its a nice helper for inspection.

## Why is it important?



To help with debugging the Elastic Agent and tests.

## Checklist




## How to test this PR locally



```
$ AGENT_COLLECT_DIAG=true mage integration:test
$ ls build/diagnostics/
```

```
$ AGENT_KEEP_INSTALLED=true mage integration:single TestInstallWithoutBasePath
$ # ssh into the host see that the Elastic Agent is still installed
$ mage integration:clean # needs to be cleaned as its now installed
```

## Related issues


- Closes #2953
